File: leetcode-algorithms/009. Palindrome Number/PalindromeNumber.cs
public class Solution 
    {
        public bool IsPalindrome(int x)
        {
            if (x < 0) return false;
            int y = x;
            double rev = 0;
            while (x != 0)
            {
                int pop = x % 10;
                x /= 10;
                //if (rev > int.MaxValue / 10 || (rev == int.MaxValue / 10 && pop > 7))
                    //return 0;
                //if (rev < int.MinValue / 10 || (rev == int.MinValue / 10 && pop < -8))
                    //return 0;
                rev = rev * 10 + pop;
            }
            if (rev == y) return true;
            else return false;





            
        }
    }

File: leetcode-algorithms/009. Palindrome Number/PalindromeNumber.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        l1 = []
        while x != 0:
            l1.append(x % 10)
            x //= 10
        l2 = l1.copy()
        l1.reverse()
        for a, b in zip(l1, l2):
            if a != b:
                return False
        return True